app.go

```diff
@@ -2,109 +2,50 @@ package main
 
 import (
 	"os"
-	"io"
-	"log"
 	"net/http"
-	"fmt"
 	"github.com/gorilla/mux"
 	"github.com/jawher/mow.cli"
-
+	fthealth "github.com/Financial-Times/go-fthealth/v1a"
+	log "github.com/Sirupsen/logrus"
+	"time"
 )
 
+const serviceName = "content-preview"
+const serviceDescription = "A RESTful API for retrieving and transforming content to preview data"
+
+
+var timeout = time.Duration(5 * time.Second)
+var client = &http.Client{Timeout: timeout}
+
 func main() {
-	app := cli.App("content-preview", "A RESTful API for retrieving and transforming content preview data")
+
+	log.SetLevel(log.InfoLevel)
+	log.Infof("%s service started with args %s", serviceName, os.Args)
+
+	app := cli.App(serviceName, serviceDescription)
 	appPort := app.StringOpt("app-port", "8084", "Default port for app")
 	mapiAuth := app.StringOpt("mapi-auth", "default", "Basic auth for MAPI")
-	mapiUri := app.StringOpt("mapi-uri", "http://methode-api-uk-p.svc.ft.com/eom-file/", "Host and path for MAPI")
+	mapiUri := app.StringOpt("mapi-uri", "http://methode-api-uk-p.svc.ft.com", "Host and port for MAPI")
 	matHostHeader := app.StringOpt("mat-host-header", "methode-article-transformer", "Hostheader for MAT")
-	matUri := app.StringOpt("mat-uri", "http://ftapp05951-lvpr-uk-int:8080/content-transform/", "Host and path for MAT")
+	matUri := app.StringOpt("mat-uri", "http://ftapp05951-lvpr-uk-int:8080", "Host and port for MAT")
 
 	app.Action = func() {
 		r := mux.NewRouter()
-		handler := Handlers{*mapiAuth, *mapiUri, *matUri, *matHostHeader}
+		handler := Handlers{*mapiAuth, *matHostHeader, *mapiUri, *matUri}
 		r.HandleFunc("/content-preview/{uuid}", handler.contentPreviewHandler)
 		r.HandleFunc("/build-info", handler.buildInfoHandler)
-		r.HandleFunc("/ping", pingHandler)
+		r.HandleFunc("/__health", fthealth.Handler(serviceName, serviceDescription, handler.mapiCheck(), handler.matCheck()))
+		r.HandleFunc("/__ping", pingHandler)
 		http.Handle("/", r)
 
-		log.Fatal(http.ListenAndServe(":" + *appPort, nil))
-
-	}
-	app.Run(os.Args)
-
-}
-
-type Handlers struct {
-	mapiAuth string
-	mapiUri string
-	matUri string
-	matHostHeader string
-}
-
-func pingHandler(w http.ResponseWriter, r *http.Request){
-
-}
-
-func (h Handlers) buildInfoHandler(w http.ResponseWriter, r *http.Request){
-
-}
-
-func (h Handlers) contentPreviewHandler(w http.ResponseWriter, r *http.Request) {
-	log.Printf("received request");
-
-	vars := mux.Vars(r)
-	uuid := vars["uuid"]
-
-	if uuid == "" {
-		log.Fatal("missing UUID")
+		log.Fatal(http.ListenAndServe(":"+*appPort, nil))
```

Comment:
I suggest wrapping this in the metrics aware and transaction id aware http handlers - they will automatically capture metrics for each http method, and these can then be exported to graphite if you want them, and will also output a request log, that includes a transaction id.

example, where r is the mux router:
http.ListenAndServe(fmt.Sprintf(":%d", *port),
			httphandlers.HTTPMetricsHandler(metrics.DefaultRegistry,
				httphandlers.TransactionAwareRequestLoggingHandler(log.StandardLogger(), r)))